### DISCUSSION TITLE
osgEarth 3.3 Warning: Material::apply messages

### DISCUSSION THREAD

**Discussion by davisjamesf** [pelicanmapping/osgearth#2194] [Support Q&A] [answered]
How are osg models with materials, for example ./data/boxman.osg, loaded in osgEarth 3.3 or Open Scene Graph that has been compiled with OSG_GL3_AVAILABLE without getting the warning message "Warning: Material::apply(State&) - not supported."  You can see this by opening boxman.osg in osgviewer.   These same kind of messages will also occur loading models in osgEarth (i.e. osgearth_shadercomp.exe --test7 ..\..\data\boxman.osg)

In Open Scene Graph, this message WILL be printed out with OSG_GL3_AVAILABLE enabled and trying to load a model with textures and materials.   I am compiling OSG with OSG_GL1_AVAILABLE and OSG_GL2_AVAILABLE disabled according to instructions I'm finding in other threads in OSG and the comments at the end of the file openscenegraph/examples/osgsimplegl3.cpp.

**Comment by gwaldron** [comment, marked as answer]
If you have a `osg::Material` attribute in your model, and you built OpenSceneGraph with `OSG_GL_FIXED_FUNCTION_AVAILABLE=OFF`, you will get this message on the console. They only way to get rid of it is to either remove the `osg::Material` attribute from your model or to change the OSG notify level.
